includes/class-wc-payments.php

```diff
@@ -771,4 +773,77 @@ public static function remove_woo_admin_notes() {
 	public static function is_network_saved_cards_enabled() {
 		return apply_filters( 'wcpay_force_network_saved_cards', false );
 	}
+
+	/**
+	 * Registers platform checkout hooks if the platform checkout feature flag is enabled.
+	 */
+	public static function maybe_register_platform_checkout_hooks() {
+		if ( WC_Payments_Features::is_platform_checkout_enabled() ) {
+			add_action( 'wc_ajax_wcpay_init_platform_checkout', [ __CLASS__, 'ajax_init_platform_checkout' ] );
+			add_filter( 'determine_current_user', [ __CLASS__, 'determine_current_user_for_platform_checkout' ] );
+			// Disable nonce checks for API calls. TODO This should be changed.
+			add_filter( 'woocommerce_store_api_disable_nonce_check', '__return_true' );
+		}
+	}
+
+	/**
+	 * Used to initialize platform checkout session.
+	 *
+	 * @return void
+	 */
+	public static function ajax_init_platform_checkout() {
+		$session_cookie_name = apply_filters( 'woocommerce_cookie', 'wp_woocommerce_session_' . COOKIEHASH );
+
+		$platform_checkout_host = defined( 'PLATFORM_CHECKOUT_HOST' ) ? PLATFORM_CHECKOUT_HOST : 'http://host.docker.internal:8090';
+		$url                    = $platform_checkout_host . '/wp-json/platform-checkout/v1/init';
+		$body                   = [
+			'user_id'              => get_current_user_id(),
+			'session_cookie_name'  => $session_cookie_name,
+			'session_cookie_value' => wp_unslash( $_COOKIE[ $session_cookie_name ] ?? '' ), // phpcs:ignore WordPress.Security.ValidatedSanitizedInput
+			'store_data'           => [
+				'store_name' => get_bloginfo( 'name' ),
+				'store_logo' => wp_get_attachment_image_src( get_theme_mod( 'custom_logo' ), 'full' )[0] ?? '',
+				'blog_id'    => Jetpack_Options::get_option( 'id' ),
+			],
+		];
+		$args                   = [
+			'url'     => $url,
+			'method'  => 'POST',
+			'timeout' => 30,
+			'body'    => wp_json_encode( $body ),
+			'headers' => [
+				'Content-Type' => 'application/json',
+			],
+		];
+
+		$response_array     = wp_remote_request( $url, $args );
+		$response_body_json = wp_remote_retrieve_body( $response_array );
+
+		Logger::log( $response_body_json );
+		wp_send_json( json_decode( $response_body_json ) );
+	}
+
+	/**
+	 * Determine the current user
+	 *
+	 * @param WP_User|int $user The user to determine.
+	 */
+	public static function determine_current_user_for_platform_checkout( $user ) {
+		if ( $user ) {
+			return $user;
+		}
+		$headers = getallheaders();
```

Comment:
Not related to renaming this function but getallheaders() is not the function we should use here. It is only available when running in apache or nginx + php-fpm 7.3+ and not in the cli. I ran into this problem running this command:
```
wp option set _wcpay_feature_platform_checkout 0
```
We should update it to use `$_SERVER['HTTP_X-WCPAY-PLATFORM-CHECKOUT-USER']` which will always work and not cause a problem in cli.

Reply:
Thanks for pointing this out! I have tested this and noticed the header arrives in the `$_SERVER` superglobal as: `HTTP_X_WCPAY_PLATFORM_CHECKOUT_USER`. I have used that for the key here and in all of the other places.